art/estimators/object_detection/tensorflow_faster_rcnn.py

```diff
@@ -0,0 +1,530 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2020
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements the task specific estimator for Faster R-CNN in TensorFlow.
+"""
+import logging
+from typing import List, Dict, Optional, Tuple, Union, TYPE_CHECKING
+
+import numpy as np
+import tensorflow as tf
+from object_detection.utils import config_util
+from object_detection.builders import model_builder
+from object_detection.utils import variables_helper
+
+from art.estimators.object_detection.object_detector import ObjectDetectorMixin
+from art.estimators.tensorflow import TensorFlowEstimator
+from art.utils import Deprecated, deprecated_keyword_arg
+from art.utils import get_file
+from art.config import ART_DATA_PATH
+
+if TYPE_CHECKING:
+    from object_detection.meta_architectures.faster_rcnn_meta_arch import FasterRCNNMetaArch
+    from tensorflow.python.framework.ops import Tensor
+    from tensorflow.python.client.session import Session
+
+    from art.config import CLIP_VALUES_TYPE, PREPROCESSING_TYPE
+    from art.defences.preprocessor.preprocessor import Preprocessor
+    from art.defences.postprocessor.postprocessor import Postprocessor
+
+logger = logging.getLogger(__name__)
+
+
+class TensorFlowFasterRCNN(ObjectDetectorMixin, TensorFlowEstimator):
+    """
+    This class implements a model-specific object detector using Faster-RCNN and TensorFlow.
+    """
+
+    @deprecated_keyword_arg("channel_index", end_version="1.5.0", replaced_by="channels_first")
+    def __init__(
+        self,
+        model: Optional["FasterRCNNMetaArch"] = None,
+        filename: Optional[str] = None,
+        url: Optional[str] = None,
+        images: Optional["Tensor"] = None,
```

Comment:
I think images cannot be None. Maybe we should remove the default here?
```suggestion
        images: "Tensor",
```

Reply:
I agree.